File: Programming EF Code First/chapter6/Chapter6/Chapter 6 VB/Fluent/DataAccess/DropCreateBreakAwayWithSeedData.vb
Imports System.Data.Entity
Imports Model

Namespace DataAccess
  Public Class DropCreateBreakAwayWithSeedData
	  Inherits DropCreateDatabaseAlways(Of BreakAwayContext)
	Protected Overrides Sub Seed(ByVal context As BreakAwayContext)
	  context.Database.ExecuteSqlCommand("CREATE INDEX IX_Lodgings_Name ON Lodgings (Name)")

	  context.Destinations.Add(New Destination With {.Name = "Great Barrier Reef"})
	  context.Destinations.Add(New Destination With {.Name = "Grand Canyon"})
	End Sub
  End Class
End Namespace

File: Programming EF Code First/chapter6/Chapter6/BAGA_DataAnnotations/DataAccess/DropCreateBreakAwayWithSeedData.cs
using System.Data.Entity;
using Model;

namespace DataAccess
{
  public class DropCreateBreakAwayWithSeedData : DropCreateDatabaseAlways<BreakAwayContext>
  {
    protected override void Seed(BreakAwayContext context)
    {
      context.Database.ExecuteSqlCommand("CREATE INDEX IX_Lodgings_Name ON Lodgings (Name)");

      context.Destinations.Add(new Destination { Name = "Great Barrier Reef" });
      context.Destinations.Add(new Destination { Name = "Grand Canyon" });
    }
  }
}